internal/alb/tg/targetgroup.go

```diff
@@ -150,6 +156,37 @@ func (controller *defaultController) reconcileTGInstance(ctx context.Context, in
 	return instance, nil
 }
 
+// resolveServiceHealthCheckPort checks if the service-port annotation is a string. If so, it tries to look up a port with the same name
+// on the service and use that port's NodePort as the health check port.
+func (controller *defaultController) resolveServiceHealthCheckPort(namespace string, serviceName string, servicePortAnnotation intstr.IntOrString, targetType string) string {
+	servicePort := servicePortAnnotation.String()
+
+	//If the annotation uses the default port ("traffic-port"), do not try to look up a port by that name.
+	if servicePortAnnotation.Type == intstr.String && servicePortAnnotation.StrVal != healthcheck.DefaultPort {
+		serviceKey := namespace + "/" + serviceName
+
+		service, err := controller.store.GetService(serviceKey)
+		if err != nil {
+			return servicePort
+		}
+
+		resolvedServicePort, err := k8s.LookupServicePort(service, servicePortAnnotation)
+		if err != nil {
+			return servicePort
+		}
+
+		if resolvedServicePort.Name == servicePortAnnotation.StrVal && resolvedServicePort.NodePort != 0 {
```

Comment:
1. the resolvedServicePort.Name == servicePortAnnotation.StrVal is unneeded, which is already expressed in k8s.LookupServicePort.

1. the logic nest order should be changed to below, since mode IP don't require nodePort
```
if targetType == elbv2.TargetTypeEnumInstance {
    if resolvedServicePort.NodePort == 0 {
        return "", errors.New("service must be type NodePort for named healthCheckPort")
    }
   return strconv.Itoa(int(resolvedServicePort.NodePort)), nil
}
return resolvedServicePort.TargetPort.String()
```